src/data/mod.rs

```diff
@@ -14,43 +16,78 @@ pub struct Context {
     pub repo: GithubRepo,
     pub releases: Vec<Release>,
     pub has_prereleases: bool,
-    pub latest_dist_release: Option<DistRelease>,
+    pub latest_release: Option<usize>,
+    pub has_artifacts: bool,
 }
 
 impl Context {
-    pub fn new(repo_url: &str, cargo_dist: bool) -> Result<Self> {
+    pub fn new(repo_url: &str, artifacts_config: &ArtifactsConfig) -> Result<Self> {
         let repo = GithubRepo::from_url(repo_url)?;
-        let (releases, has_prereleases, latest_dist_release) =
-            Self::fetch_all_releases(&repo, cargo_dist)?;
+        let (releases, has_prereleases, has_artifacts, latest_release) =
+            Self::fetch_all_releases(&repo, artifacts_config)?;
+
         Ok(Self {
             repo,
             releases,
             has_prereleases,
-            latest_dist_release,
+            has_artifacts,
+            latest_release,
         })
     }
 
+    /// Get the latest release, if it exists
+    pub fn latest(&self) -> Option<&Release> {
+        self.latest_release.and_then(|idx| self.releases.get(idx))
+    }
+
+    /// Mutably get the latest release, if it exists
+    pub fn latest_mut(&mut self) -> Option<&mut Release> {
+        self.latest_release
+            .and_then(|idx| self.releases.get_mut(idx))
+    }
+
+    /// Fetch and process all the Github Releases
+    ///
+    /// Returned values are:
+    ///
+    /// * list of releases (newest to oldest)
+    /// * whether there are any prereleases (enables more complex UI)
+    /// * whether there are any artifacts (enables more complex UI)
+    /// * the release that should be considered "latest" (shown on the front page)
     #[allow(clippy::unnecessary_unwrap)]
     pub fn fetch_all_releases(
         repo: &GithubRepo,
-        cargo_dist: bool,
-    ) -> Result<(Vec<Release>, bool, Option<DistRelease>)> {
+        artifacts_config: &ArtifactsConfig,
+    ) -> Result<(Vec<Release>, bool, bool, Option<usize>)> {
```

Comment:
this return type is starting to look like it wants to be a struct... 2 bools seems like it'll get confusing

Reply:
yeah this function is a disaster